source/ada/lsp-ada_handlers.adb

```diff
@@ -613,6 +613,90 @@ package body LSP.Ada_Handlers is
       return Response;
    end On_Declaration_Request;
 
+   -------------------------------
+   -- On_Implementation_Request --
+   -------------------------------
+
+   overriding function On_Implementation_Request
+     (Self    : access Message_Handler;
+      Request : LSP.Messages.Server_Requests.Implementation_Request)
+      return LSP.Messages.Server_Responses.Location_Response
+   is
+      use Libadalang.Analysis;
+
+      Position   : LSP.Messages.TextDocumentPositionParams renames
+        Request.params;
+      Response   : LSP.Messages.Server_Responses.Location_Response
+        (Is_Error => False);
+      Imprecise  : Boolean := False;
+
+      Document : constant LSP.Ada_Documents.Document_Access :=
+        Get_Open_Document (Self, Position.textDocument.uri);
+
+      procedure Resolve_In_Context (C : Context_Access);
+      --  Utility function to gather results on one context
+
+      ------------------------
+      -- Resolve_In_Context --
+      ------------------------
+
+      procedure Resolve_In_Context (C : Context_Access) is
+         Name_Node      : constant Name := LSP.Lal_Utils.Get_Node_As_Name
+             (C.Get_Node_At (Document, Position));
+         Definition     : Defining_Name;
+         Loop_Count     : Natural := 0;
+         This_Imprecise : Boolean;
+
+      begin
+         if Name_Node = No_Name then
+            return;
+         end if;
+
+         --  Find the definition
+         Definition := Resolve_Name (Name_Node, Self.Trace, This_Imprecise);
+         Imprecise := Imprecise or This_Imprecise;
+
+         --  If we didn't find a definition, give up for this context
+         if Definition = No_Defining_Name then
+            return;
+         end if;
+
+         --  Now that we have a definition, list all the implementations for
```

Comment:
Looking at https://stackoverflow.com/questions/58380425/whats-the-difference-between-textdocument-definition-textdocument-implementati, I think that we should implement the same mechanism as for textDocument/definition here (i.e: list base and overriding subprograms).

Reply:
Allright! let’s do this as part of a subsequent PR, if OK with you?

Reply:
Yes, np for me